src/components/PopupBase/PopupBase.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import PopperJS from 'popper.js';
+import { POPUP_PLACEMENTS, ESCAPE_KEY } from '../../constants';
+
+class PopupBase extends React.Component {
+    static checkDOMPath(domElem, checkIsTarget) {
+        let currentElem = domElem;
+        while (currentElem && currentElem !== document) {
```

Comment:
I don't quite understand why exactly `while` loop is used here. It can be replaced with `if` since it's called with a single `domElem` each time or I'm missing something?

Reply:
it is called with a single element, but then it cycles through all its parents and checks if that is the one we are targeting. E.g. there is an `a` in a `p` inside the popup component. When clicked, we don't want to close the popup. In order to see that it was indeed this popup that was clicked we cycle through all the parents and apply the `checkIsTarget` to them.